mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -2729,6 +2737,10 @@ void AsmPrinter::Impl::printNamedAttribute(NamedAttribute attr) {
     return;
 
   os << " = ";
+  if (printNamedAttrFn && succeeded(printNamedAttrFn(attr))) {
+    /// If we print via the `printNamedAttrFn` callback skip printing.
```

Comment:
> /// If we print via the `printNamedAttrFn` callback **skip printing**.

Note that this isn't accurate - "printing" happens regardless. What changes is the method that's used for printing.

Reply:
Updated comment.